### PR TITLE
[Feature] Add inferencer for lidar-based segmentation

### PR DESCRIPTION
**Pull request by Xiangxu-0103**
Thanks for your contribution and we appreciate it a lot. The following instructions would make your pull request more healthy and more easily get feedback. If you do not understand some items, don't worry, just make the pull request and seek help from maintainers.

## Motivation

Please describe the motivation of this PR and the goal you want to achieve through this PR.

## Modification

Please briefly describe what modification is made in this PR.

## BC-breaking (Optional)

Does the modification introduce changes that break the back-compatibility of the downstream repos?
If so, please describe how it breaks the compatibility and how the downstream projects should modify their code to keep compatibility with this PR.

## Use cases (Optional)

If this PR introduces a new feature, it is better to list some use cases here, and update the documentation.

## Checklist

1. Pre-commit or other linting tools are used to fix the potential lint issues.
2. The modification is covered by complete unit tests. If not, please add more unit test to ensure the correctness.
3. If the modification has potential influence on downstream projects, this PR should be tested with downstream projects.
4. The documentation has been modified accordingly, like docstring or example tutorials.

### CODE REVIEW

File: mmdet3d/apis/inferencers/lidar_seg3d_inferencer.py [hunk 1]
```diff
@@ -0,0 +1,199 @@
...
+InputType = Union[str, np.ndarray]
+InputsType = Union[InputType, Sequence[InputType]]
+PredType = Union[InstanceData, InstanceList]
+ImgType = Union[np.ndarray, Sequence[np.ndarray]]
+ResType = Union[Dict, List[Dict], InstanceData, List[InstanceData]]
+
+
+@INFERENCERS.register_module(name='seg3d-lidar')
+@INFERENCERS.register_module()
+class LidarSeg3DInferencer(BaseDet3DInferencer):
```
Comment (ZCMax) [inline mmdet3d/apis/inferencers/lidar_seg3d_inferencer.py:25]: It's a little bit strange to inherit from `BaseDet3DInferencer`, should we add a `BaseSeg3DInferencer`?

Comment (ZCMax) [reply mmdet3d/apis/inferencers/lidar_seg3d_inferencer.py:25]: Probably we will support `VisionSeg3DInferencer` in the future

File: configs/pointnet2/metafile.yml [hunk 2]
```diff
@@ -17,7 +17,7 @@ Collections:
 Models:
   - Name: pointnet2_ssg_2xb16-cosine-200e_scannet-seg-xyz-only
     In Collection: PointNet++
-    Config: configs/pointnet/pointnet2_ssg_2xb16-cosine-200e_scannet-seg-xyz-only.py
```
Comment (ZwwWayne) [inline configs/pointnet2/metafile.yml:20]: We should fix this in another PR.

File: mmdet3d/apis/inferencers/lidar_seg3d_inferencer.py [hunk 3]
```diff
@@ -0,0 +1,195 @@
...
```
Comment (JingweiZhang12) [inline mmdet3d/apis/inferencers/lidar_seg3d_inferencer.py:25]: Suggest we can rename `BaseDet3DInferencer` as `Base3DInferencer`

File: mmdet3d/apis/inferencers/lidar_seg3d_inferencer.py [hunk 4]
```diff
@@ -0,0 +1,195 @@
...
+    }
+    postprocess_kwargs: set = {
+        'print_result', 'pred_out_file', 'return_datasample'
+    }
+
+    def __init__(self,
+                 model: Union[ModelType, str, None] = None,
+                 weights: Optional[str] = None,
+                 device: Optional[str] = None,
+                 scope: Optional[str] = 'mmdet3d',
```
Comment (JingweiZhang12) [inline mmdet3d/apis/inferencers/lidar_seg3d_inferencer.py:60]:
```suggestion
                 scope: str = 'mmdet3d',
```

**Review by JingweiZhang12 — APPROVED** [review]

**Review by ZCMax — APPROVED** [review]

**Pull request merged**